earth2studio/data/arco.py

```diff
@@ -109,101 +112,115 @@ def __call__(
         # Make sure input time is valid
         self._validate_time(time)
 
-        # Fetch index file for requested time
-        data_arrays = []
-        for t0 in time:
-            data_array = self.fetch_arco_dataarray(t0, variable)
-            data_arrays.append(data_array)
+        xr_array = asyncio.run(
+            asyncio.wait_for(self.create_data_array(time, variable), self.async_timeout)
+        )
 
         # Delete cache if needed
         if not self._cache:
             shutil.rmtree(self.cache)
 
-        return xr.concat(data_arrays, dim="time")
+        return xr_array
 
-    def fetch_arco_dataarray(
-        self,
-        time: datetime,
-        variables: list[str],
+    async def create_data_array(
```

Comment:
async code aways confuses me haha. Maybe I am missing something but does the create data array also need to be async. Could you not have the async part only be around the for loop for fetching the data.

Reply:
The trouble with async is that you cant really "await" inside a non-async function. The async generator here needs an await inside of it. So I find it easiest to rather enter entire function into async rather than do it through the loop. Rather than having a pool of tasks you have to collect, you can just have one async entry point here.

Perhaps there is a more parsimonious way to do this, but with the async generator its hard do something that looks more like a threaded solution.

Note that the important function `fetch_array` isnt async in this case.

Reply:
Also this code is very confusing, because it has the throttling control to limit number of async processes. Its by no means a "strandard" async usage you would find else where.